### DISCUSSION TITLE
Keychain Access Issue? (Mac OSX)

### DISCUSSION THREAD

**Discussion by thewatts** [rubycdp/cuprite#195] [General]
Hello! :) 

Thanks so much for all your hard work on this project!

I'm having a bit of an issue when trying to launch Cuprite in debug mode - it opens the browser, but it's blank, and this error is output to the terminal:

![image](https://user-images.githubusercontent.com/1141717/168439841-abc42900-b3ba-49ca-9848-03a0e7ac0d65.png)

```
[0514/110457.744445:ERROR:developer_id_certificate_reauthorize.mm(92)] SecKeychainFindGenericPassword (main): Error Domain=NSOSStatusErrorDomain Code=-25293 "errKCAuthFailed / errSecAuthFailed:  / Authorization/Authentication failed." (-25293)
[0514/110457.745184:ERROR:developer_id_certificate_reauthorize.mm(93)] SecKeychainFindGenericPassword (backup): Error Domain=NSOSStatusErrorDomain Code=-25300 "errKCItemNotFound / errSecItemNotFound:  / The item cannot be found." (-25300)
[74400:259:0514/110457.802452:ERROR:keychain_password_mac.mm(103)] Keychain lookup failed: Error Domain=NSOSStatusErrorDomain Code=-25308 "errKCInteractionNotAllowed / errSecInteractionNotAllowed:  / Interaction is not allowed with the Security Server." (-25308)
objc[74409]: Class WebSwapCGLLayer is implemented in both /System/Library/Frameworks/WebKit.framework/Versions/A/Frameworks/WebCore.framework/Versions/A/Frameworks/libANGLE-shared.dylib (0x2220f31a8) and /Applications/Google Chrome.app/Contents/Frameworks/Google Chrome Framework.framework/Versions/101.0.4951.64/Libraries/libGLESv2.dylib (0x1063e53e8). One of the two will be used. Which one is undefined.
```

I've Googled around for what this could be, but have come up short.

Thanks for your help!

**Comment by thewatts** [comment]
Additionally - 

I tried Chromium - I still get a blank page.

While I don't get the keychain errors with Chromium, I still get the last line:

```
objc[74409]: Class WebSwapCGLLayer is implemented in both /System/Library/Frameworks/WebKit.framework/Versions/A/Frameworks/WebCore.framework/Versions/A/Frameworks/libANGLE-shared.dylib (0x2220f31a8) and /Applications/Google Chrome.app/Contents/Frameworks/Google Chrome Framework.framework/Versions/101.0.4951.64/Libraries/libGLESv2.dylib (0x1063e53e8). One of the two will be used. Which one is undefined.
```

**Comment by route** [comment]
Hi! I'll check when I get back home I'm traveling. Mind sharing mac os details as well? So that we are on the same page

**Reply by thewatts** [reply]
Thanks, @route!

- Computer is an M1 Mac Mini (8gb)
- macos 12.3.1

**Comment by peterberkenbosch** [comment]
Experiencing the same here. M1 on macos 12.3.1
Debug output in a gist here: https://gist.github.com/peterberkenbosch/c10b6202bdc14b561e17339a9a838fc0

I also have Brave installed and that's my default browser, would that cause any conflicts?

If it helps I have an opensource template app that I use with this: https://github.com/peterberkenbosch/rails-starter-template when I add `page.driver.debug` on the system spec the page does open in Chrome but I have a blank page and the error as listed in the gist above.

**Reply by peterberkenbosch** [reply]
gentle ping @route
